dmscripts/generate_framework_agreement_signature_pages.py

```diff
@@ -95,28 +114,66 @@ def render_html_for_suppliers_awaiting_countersignature(rows, framework, templat
         data['countersigned_at'] = datetime.strptime(
             data['countersigned_at'], '%Y-%m-%dT%H:%M:%S.%fZ'
         ).strftime('%d %B %Y')
+        if data['signed_agreement_returned_at']:
+            data['signed_agreement_returned_at'] = datetime.strptime(
+                data['signed_agreement_returned_at'], '%Y-%m-%dT%H:%M:%S.%fZ'
+            ).strftime('%d %B %Y')
         data['includeCountersignature'] = True
-        html = render_html(template_path, data)
-        save_page(html, data['supplier_id'], output_dir, "agreement-countersignature")
-    shutil.copyfile(template_css_path, os.path.join(output_dir, 'framework-agreement-signature-page.css'))
-    shutil.copyfile(
-        template_countersignature_path,
-        os.path.join(output_dir, 'framework-agreement-countersignature.png')
-    )
+
+        html_dir = output_dir / str(data['supplier_id'])
+        output_pages = []
+        for html_page in html_pages:
+            html = render_html(str(html_page), data)
+            page_path = save_page(html, data['supplier_id'], html_dir, html_page.stem)
+            output_pages.append(Path(page_path).resolve())
+
+        for static_file in static_files:
+            shutil.copyfile(static_file, Path(html_dir, static_file.name))
```

Comment:
It occured to me that we can save a lot of space and copy operations by using symlinks for static files:

```suggestion
            Path(html_dir, static_file.name).symlink_to(static_file.resolve())
```

What do you think?

Reply:
I don't see why not, are we sure this will have the same behaviour on Jenkins?

Reply:
It should do, I don't think there's anything special about the Jenkins filesystem in this regard.